build.sbt

```diff
@@ -343,3 +345,15 @@ lazy val `commons-akka` = project
       "com.typesafe.akka" %% "akka-testkit" % akkaVersion % Test,
     ),
   )
+
+lazy val `commons-okhttp` = project
+  .dependsOn(`commons-core` % CompileAndTest)
+  .dependsOn(`commons-jetty` % CompileAndTest)
```

Comment:
I'm not sure that it works exactly as I expect. I would like to add commons-jetty dependency only for tests.

Reply:
`CompileAndTest` is used so that test code of one module can depend on test code of other module. By default the dependency is only established on production code.
If you want a standard test-only dependency, then `% Test` is what you want.

However, I'm a little afraid that we're going to have too many micro modules in `commons` with a separate one created for every dependency. I'm thinking if it would be better to keep one module for various external library integrations and make its external dependencies `Optional`. Currently I have no idea which solution would be better. Both are ugly in some way. WDYT?

Reply:
If we declare external dependencies ```Optional```, it will limit micromodules, but also enforce the user to import them manually. I don't want to change the behavior of other modules, so I decided to create the new one, but ofc your comment is correct. I'm going to merge commons-jetty with commons-okhttp to see the difference and possibly update the PR.

Reply:
Please take a look. I've merged commons-jetty and commons-okhttp into commons-http module. I've fixed parameter encoding, because okHttp doesn't encode + as space in path parameter (Hardcoded plusIsSpace=false parameter in HttpUrl.push method). If you accept these changes, I will correct the documentation.

Reply:
Optional dependencies in case of integration modules like these seem OK to me because there's no way to use e.g. `RestClient` for `OkHttpClient` without creating `OkHttpClient` first. The programmer actually needs to use that dependency himself so it seems OK for me to expect that he will declare it explicitly.

Optional dependency would be very ugly if it was an _internal_ dependency of a `commons-something` module. For example, Akka is an internal dependency of `commons-redis`. It uses Akka, but it (almost) doesn't expose it to the user of Redis driver. So that's a very different situation from what we have in `commons-jetty` etc.

I think we should go all or nothing with merging all integration modules into one. IMHO, that would include `commons-akka`, `commons-jetty`, `commons-spring`, `commons-kafka` and possibly `commons-mongo`. I don't expect you to do it, so you may as well leave `commons-okhttp` for now and some day I'll refactor the project structure.

Reply:
Ok, I see. If I have enough time I will prepare PR with merging these modules into one. This OkHttp integration can wait. It could be also a great opportunity for me to clean up this commons-kafka module, because it is outdated and some things should be rewritten.